### ISSUE TITLE
DeviceInfoMesg.getProduct() is null

### ISSUE THREAD

**Issue by chris-obermeier** [sumoDave/GarminDiveFitAppender#5]
Found one more bug when combining my other files.  For some reason this group of .fit files is returning a null on your getProduct() method.

Files are attached in .zip.
[Fit Files.zip](https://github.com/user-attachments/files/15974367/Fit.Files.zip)

Error message is as follows:

```
Processing file: C:\Users\ChrisObermeier\Desktop\Fit Files\Fit Files\15898242791_ACTIVITY.fit
Processing file: C:\Users\ChrisObermeier\Desktop\Fit Files\Fit Files\15898243294_ACTIVITY.fit
Processing file: C:\Users\ChrisObermeier\Desktop\Fit Files\Fit Files\15898243882_ACTIVITY.fit
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.lang.Integer.toString()" because the return value of "com.garmin.fit.DeviceInfoMesg.getProduct()" is null
        at DeviceInfoMessage.filterMessages(DeviceInfoMessage.java:14)
        at DiveProfileData.initialise(DiveProfileData.java:73)
        at DiveProfile.initialiseMessages(DiveProfile.java:15)
        at Main.main(Main.java:28)
```

**Comment by sumoDave** [comment]
Thanks for the files. 
I was using the Product Id and Device Index as a composite key to identify unique entries.
Apparently not all devices have a product id.
Should be a quick fix.

**Comment by chris-obermeier** [comment]
No worries!  If you ever need a second pair of eyes let me know!


Thanks,

Chris Obermeier

Head of Engineering

[Logo  Description automatically generated]

—

Remote

Email: ***@***.******@***.***>

Website: xpanse.com

Phone: 414.429.4185



[A picture containing company name  Description automatically generated]    [A picture containing icon  Description automatically generated]

________________________________
From: sumoDave ***@***.***>
Sent: Tuesday, June 25, 2024 6:29:28 PM
To: sumoDave/GarminDiveFitAppender ***@***.***>
Cc: Chris Obermeier ***@***.***>; Author ***@***.***>
Subject: **EXT**Re: [sumoDave/GarminDiveFitAppender] DeviceInfoMesg.getProduct() is null (Issue #5)


Thanks for the files.
I was using the Product Id and Device Index as a composite key to identify unique entries.
Apparently not all devices have a product id.
Should be a quick fix.

**Comment by sumoDave** [comment]
Fixed in v2.1.0
